src/case/mod.rs

```diff
@@ -0,0 +1,56 @@
+mod search;
```

Comment:
all this stuff can be much simplified if i give up on trying to reuse the search.rs file in both the generator and generatee (which is worth doing). like, that's the reason it's 3 files here, it should just be 1.